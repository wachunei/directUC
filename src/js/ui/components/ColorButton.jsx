import PropTypes from "prop-types";
import styled, { css } from "styled-components";
import { darken } from "polished";
import { COLORS } from "../theme";

const CircleButton = styled.button`
  display: inline-flex;
  appearance: none;
  height: 45px;
  width: 45px;
  background: ${(props) => COLORS[props.value]};
  font-size: 24px;
  border-radius: 50%;
  transition: all 0.1s ease-in-out;
  cursor: pointer;
  margin: 5px 10px 5px 0px;
  border-style: solid;
  border-width: 2px;
  border-color: transparent;
  overflow: hidden;
  line-height: 0;
  align-items: center;
  justify-content: center;

  span {
    display: none;
  }

  ${(props) =>
    props.selected &&
    css`
      border-color: ${darken(0.1, COLORS[props.value])};
    `};

  &:disabled {
    cursor: not-allowed;
    filter: grayscale(100%) opacity(80%);
  }

  &:focus {
    outline-width: 2px;
    outline-color: ${(props) => darken(0.1, COLORS[props.value])};
  }
  &:last-child {
    margin-right: 0;
  }
`;

const ColorButton = ({ label, emoji, ...props }) => (
  <CircleButton {...props}>
    <span>{label}</span>
    {emoji}
  </CircleButton>
);

ColorButton.propTypes = {
  label: PropTypes.string.isRequired,
  emoji: PropTypes.string.isRequired,
};

export default ColorButton;
