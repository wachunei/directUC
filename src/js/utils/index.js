export const removeSuffix = (suffix, value) =>
  value.endsWith(suffix) ? value.slice(0, -suffix.length) : value;

export const parseDistinguishedName = (name) =>
  name.replace(
    /\w\S*/g,
    (txt) => txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase()
  );

export const isFirefox = !window.chrome.app;
