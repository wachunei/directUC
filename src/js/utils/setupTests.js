// eslint-disable-next-line import/no-extraneous-dependencies
import "jest-styled-components";
