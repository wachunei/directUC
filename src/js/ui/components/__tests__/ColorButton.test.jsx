import ColorButton from "../ColorButton";
import { render, getElement } from "./utils";

describe("Button Component", () => {
  test("default should match snapshot", () => {
    const { container } = render(
      <ColorButton
        type="button"
        name="primaryColor"
        value="blue"
        label="Azul"
        emoji="ðŸ”µ"
        onClick={jest.fn}
        selected
        disabled={false}
      />
    );
    expect(getElement(container)).toMatchSnapshot();
  });
});
