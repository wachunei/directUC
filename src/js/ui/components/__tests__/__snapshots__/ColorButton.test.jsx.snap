// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component default should match snapshot 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  height: 45px;
  width: 45px;
  background: #1F95EB;
  font-size: 24px;
  border-radius: 50%;
  -webkit-transition: all 0.1s ease-in-out;
  transition: all 0.1s ease-in-out;
  cursor: pointer;
  margin: 5px 10px 5px 0px;
  border-style: solid;
  border-width: 2px;
  border-color: transparent;
  overflow: hidden;
  line-height: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-color: #127ac5;
}

.c0 span {
  display: none;
}

.c0:disabled {
  cursor: not-allowed;
  -webkit-filter: grayscale(100%) opacity(80%);
  filter: grayscale(100%) opacity(80%);
}

.c0:focus {
  outline-width: 2px;
  outline-color: #127ac5;
}

.c0:last-child {
  margin-right: 0;
}

<button
  class="c0"
  name="primaryColor"
  type="button"
  value="blue"
>
  <span>
    Azul
  </span>
  ðŸ”µ
</button>
`;
